HW06.Rmd

```diff
@@ -1,92 +1,147 @@
+
+
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+
+
+#print out the t-test results
+```
+
+
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+
+tidy_data <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c ("group", "order"), convert = TRUE)
+
+
+#create a graphing function
+graph <- function(x) {
+  gene_data <- tidy_data %>%
+    filter(gene_symbol == gene)
```

Comment:
I gather from your codes that you managed to run the graph function but not the t-test. I was just commenting on Ben's codes previously that my issue was the opposite! This may be repetitive as I wrote the same thing for Ben but your codes here before the graph function are similar to mine, except that I put it before initiating t-test. I decided to reorganize the file first and filter gene before doing a for loop to run a t-test on each gene and that helped me. Maybe this can be something for you to look into. The only issue with my way of doing is that somehow my "gene" in for (gene in genes_of_interest) gets assigned as a variable and I think that led to why my second for loop is not working, thus I can't get my graphs. :( Still trying to figure this part out!